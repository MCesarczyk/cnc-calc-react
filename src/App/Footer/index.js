import { StyledFooter, FooterContent } from "./styled";

const Footer = ({ content }) => (
  <StyledFooter>
    <FooterContent>
      {content}
    </FooterContent>
  </StyledFooter>
);

export default Footer;