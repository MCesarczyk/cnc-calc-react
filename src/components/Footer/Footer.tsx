import { AUTHOR_PAGE_ADDRESS, AUTHOR_PAGE_LABEL, DATE } from 'assets/fixtures/author';
import styled from 'styled-components';

export const Footer = () => (
  <FooterWrapper>
    <FooterContent>
      Copyright &copy; {DATE} by&nbsp;
      <FooterLink href={AUTHOR_PAGE_ADDRESS}>{AUTHOR_PAGE_LABEL}</FooterLink>
    </FooterContent>
  </FooterWrapper>
);

const FooterWrapper = styled.footer`
  text-align: center;
  margin: 0;
`;

const FooterContent = styled.p`
  color: ${({ theme }) => theme.color.primary};
  padding: 0 25px;
  font-size: 14px;
`;

const FooterLink = styled.a`
  color: inherit;
  transition: 0.5s ease-out;

  &:hover {
    color: crimson;
  }
`;
