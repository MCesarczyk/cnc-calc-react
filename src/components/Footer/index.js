import Link from "../Link";
import { StyledFooter, FooterContent } from "./styled";

const Footer = () => (
  <StyledFooter>
    <FooterContent>
      Copyright &copy; {new Date().getFullYear()} by&nbsp;
      <Link
        adress={"mailto:michal.cesarczyk@gmail.com"}
        text="MichaÅ‚&nbsp;Cesarczyk"
      />
    </FooterContent>
  </StyledFooter>
);

export default Footer;