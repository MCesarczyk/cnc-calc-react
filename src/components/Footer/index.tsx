import { AUTHOR_PAGE_ADDRESS, AUTHOR_PAGE_LABEL, DATE } from "../../assets/fixtures/author";
import { FooterWrapper, FooterContent, FooterLink } from "./styled";

const Footer = () => (
  <FooterWrapper>
    <FooterContent>
      Copyright &copy; {DATE} by&nbsp;
      <FooterLink
        href={AUTHOR_PAGE_ADDRESS}
      >
        {AUTHOR_PAGE_LABEL}
      </FooterLink>
    </FooterContent>
  </FooterWrapper>
);

export default Footer;