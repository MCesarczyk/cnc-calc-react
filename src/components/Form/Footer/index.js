import { useContext } from "react";
import LanguageContext from "../../../features/language/context";
import languages from "../../../assets/fixtures/languages";
import { ButtonsContainer } from "./styled";
import { FormButton } from "../Button";

const FormFooter = () => {
    const { langId } = useContext(LanguageContext);

    return (
        <ButtonsContainer>
            <FormButton type="submit">
                {languages[langId]?.buttonLabel.submit}
            </FormButton>
            <FormButton type="reset">
                {languages[langId]?.buttonLabel.reset}
            </FormButton>
        </ButtonsContainer>
    )
};

export default FormFooter;