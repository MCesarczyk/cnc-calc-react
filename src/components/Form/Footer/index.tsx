import { ButtonsContainer, FooterButton } from "./styled";

interface FormFooterProps {
  buttons: {
    id: number;
    label: string;
    type: "submit" | "reset";
  }[];
}

const FormFooter = ({ buttons }: FormFooterProps) => (
  <ButtonsContainer>
    {buttons.map((button) => (
      <FooterButton key={button.id} type={button.type}>
        {button.label}
      </FooterButton>
    ))}
  </ButtonsContainer>
);

export default FormFooter;
