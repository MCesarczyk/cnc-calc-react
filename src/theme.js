export const theme = {
    color: {
        primary: "#1e3246",
        background: "#133246",
    },
    breakpoint: {
        mobileMax: "480px",
    },
}