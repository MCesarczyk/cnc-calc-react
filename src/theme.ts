export const theme = {
  color: {
    primary: '#1e3246',
    primaryLight: '#2e4c6b',
    primaryText: '#000',
    contrastText: '#fff',
    background: '#133246',
    elementBackground: '#fff',
    submitButtonText: '#19aa6e',
    resetButtonText: '#dc3223',
    copyButtonText: '#dc143c',
    activeNavlinkText: '#2188FF',
    invalid: 'beige',
    invalidBackground: 'beige',
    disabled: '#ddd',
    disabledText: '#808080',
    disabledBackground: '#ddd',
    placeholder: '#747474',
  },
  breakpoint: {
    xs: '480px',
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px',
    xxl: '1400px',
  },
};
