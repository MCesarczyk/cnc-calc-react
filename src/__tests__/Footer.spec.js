import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import { theme } from "../theme";
import Footer from "../components/Footer";
import "jest-styled-components";

test("Footer rendered properly", () => {
  const component = renderer.create(
    <ThemeProvider theme={theme} >
      <Footer />
    </ThemeProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});