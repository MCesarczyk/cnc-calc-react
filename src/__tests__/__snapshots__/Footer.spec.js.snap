// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer rendered properly 1`] = `
.c2 {
  color: inherit;
}

.c0 {
  margin: 30px 0px;
}

.c1 {
  color: #1e3246;
  padding: 0 25px;
  font-size: 14px;
}

<footer
  className="c0"
>
  <p
    className="c1"
  >
    Copyright © 
    2022
     by 
    <a
      className="c2"
      href="mailto:michal.cesarczyk@gmail.com"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      Michał Cesarczyk
    </a>
  </p>
</footer>
`;
